### PR TITLE
vs-eng-Integrate-procedural-anims-to-an-attack-LV-403

### PR DESCRIPTION
**Pull request by troberts2**
I am merging a different branch because I don't know what went wrong with my original one but it turned into a disaster. Moral of the story is this one has everyones updates and is not bugged hopefully.

### CODE REVIEW

File: Assets/GameAssets/Boss/BossScripts/LookAtMoveDirection.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+using UnityEngine;
```
Comment (ASLH68) [inline Assets/GameAssets/Boss/BossScripts/LookAtMoveDirection.cs:1]: FileHeader

File: Assets/GameAssets/Boss/BossScripts/LookAtMoveDirection.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+public class LookAtMoveDirection : MonoBehaviour
```
Comment (ASLH68) [inline Assets/GameAssets/Boss/BossScripts/LookAtMoveDirection.cs:3]: Class XML Comment

**Review by ASLH68 — CHANGES_REQUESTED** [review]

File: Assets/GameAssets/Boss/BossScripts/LookAtMoveDirection.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+{
+    private Vector3 lastPosition; // Store the last frame's position
+    [SerializeField] private float rotationSpeed = 10f; // Speed at which the object rotates
```
Comment (AlecatEnjoyer) [inline Assets/GameAssets/Boss/BossScripts/LookAtMoveDirection.cs:6]: Make the comment on the side a tooltip

**Review by AlecatEnjoyer — APPROVED** [review]
I approve just left a small comment

File: Assets/GameAssets/Boss/BossScripts/Attacks/Lockdown/LockdownAttackController.cs [hunk 4]
```diff
@@ -96,7 +96,6 @@ protected override void BeginAttack()
     {
         // This is a passive attack this ends when it's scene is over
         // This should only subscribe during its lifetime as it's waiting for it's scene to end
```
Comment (RyanASwanson) [inline Assets/GameAssets/Boss/BossScripts/Attacks/Lockdown/LockdownAttackController.cs:98]: Remove these comments as the line they are associated with are gone

File: Assets/GameAssets/Boss/BossScripts/LookAtMoveDirection.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    /// <summary>
+    /// changes the rotation of the the object to look at the direction it is moving toward
+    /// </summary>
+    private void Update()
+    {
+        // Calculate the velocity (difference in position over time)
+        Vector3 velocity = (transform.position - lastPosition) / Time.deltaTime;
+
+        // Update the last position for the next frame
+        lastPosition = transform.position;
```
Comment (RyanASwanson) [inline Assets/GameAssets/Boss/BossScripts/LookAtMoveDirection.cs:26]: Maybe we could store a Transform so we don't have to update it every frame? No clue if that would be more efficient tho

**Review by RyanASwanson — COMMENTED** [review]
Left some minor comments. Nothing that requires blocking this

**Pull request merged**